spring-integration-file/src/main/java/org/springframework/integration/file/remote/RemoteFileTemplate.java

```diff
@@ -303,8 +304,6 @@ public String send(Message<?> message, String subDirectory, FileExistsMode... mo
 
 	private String send(Message<?> message, String subDirectory, FileExistsMode mode) {
 		Assert.notNull(this.directoryExpressionProcessor, "'remoteDirectoryExpression' is required");
-		Assert.isTrue(!FileExistsMode.APPEND.equals(mode) || !this.useTemporaryFileName,
-				"Cannot append when using a temporary file name");
```

Comment:
I suggest to make a warning message in the `doInit()` when this condition is `true`.
We still gong to ignore, but users would know that it does not make sense to use both `useTemporaryFileName` and `FileExistsMode.APPEND`

Reply:
I've added a warning message in `doInit()` when `APPEND` mode is configured with `useTemporaryFileName` is `true`. Thanks for the good suggestion!